api/tests/test_game_state_server.py

```diff
@@ -1,288 +1,296 @@
     )
 
 
 @pytest.mark.asyncio
-async def test_delete_token(gss_with_client):
-    await async_collect(
-        gss_with_client.updates_received(
-            [VALID_ACTION], TEST_ROOM_ID, TEST_CLIENT_ID, TEST_REQUEST_ID
-        )
-    )
-    reply = await async_collect(
-        gss_with_client.updates_received(
-            [DeleteAction(action='delete', data=VALID_TOKEN.id)],
-            TEST_ROOM_ID,
-            TEST_CLIENT_ID,
-            TEST_REQUEST_ID,
-        )
-    )
-    assert_all_match(
-        reply,
-        [{'contents': {'type': 'state', 'data': [], 'request_id': TEST_REQUEST_ID}}],
+async def test_delete_token(gss: GameStateServer) -> None:
+    responses = await collect_responses(
+        gss,
+        requests=[
+            DecoratedRequest('create-request-id', [VALID_ACTION]),
+            DecoratedRequest(
+                'delete-request-id',
+                [DeleteAction(action='delete', data=VALID_TOKEN.id)],
+            ),
+        ],
+        response_count=3,
     )
+    assert responses == [
+        Response('connected', []),
+        Response('state', [VALID_TOKEN], 'create-request-id'),
+        Response('state', [], 'delete-request-id'),
+    ]
 
 
 @pytest.mark.asyncio
-async def test_delete_non_existent_token(gss_with_client):
-    reply = await async_collect(
-        gss_with_client.updates_received(
-            [DeleteAction(action='delete', data=VALID_TOKEN.id)],
-            TEST_ROOM_ID,
-            TEST_CLIENT_ID,
-            TEST_REQUEST_ID,
-        )
+async def test_delete_non_existent_token(gss: GameStateServer) -> None:
+    responses = await collect_responses(
+        gss,
+        requests=[
+            DecoratedRequest(
+                'delete-request-id',
+                [DeleteAction(action='delete', data='unused-token-id')],
+            )
+        ],
+        response_count=3,
     )
-    assert_all_match(
-        reply, [{'contents': {'type': 'error'}}, {'contents': {'type': 'state'}}]
+
+    assert states(responses) == [Response('state', [], 'delete-request-id')]
+    assert_matches(
+        errors(responses), [{'type': 'error', 'request_id': 'delete-request-id'}]
     )
 
 
 @pytest.mark.asyncio
-async def test_delete_after_reload(gss_with_client):
-    await async_collect(
-        gss_with_client.updates_received(
-            [VALID_ACTION], TEST_ROOM_ID, TEST_CLIENT_ID, 'initial_request_id'
-        )
-    )
-    await gss_with_client.connection_dropped(TEST_CLIENT_ID, TEST_ROOM_ID)
-    await gss_with_client.new_connection_request(
-        TEST_CLIENT_ID, '127.0.0.1', TEST_ROOM_ID
-    )
-    reply = await async_collect(
-        gss_with_client.updates_received(
-            [DeleteAction(action='delete', data=VALID_TOKEN.id)],
-            TEST_ROOM_ID,
-            TEST_CLIENT_ID,
-            TEST_REQUEST_ID,
-        )
+async def test_delete_after_reload(gss: GameStateServer) -> None:
```

Comment:
I think this should be removed, as it's just testing deleting tokens again.

Reply:
Makes sense, can we do that afterwards so I don't have to rebase all of the other prs?